django/db/models/fields/related.py

```diff
@@ -2051,6 +2052,17 @@ def db_type(self, connection):
     def db_parameters(self, connection):
         return {"type": self.db_type(connection), "check": []}
 
+    def from_db_value(self, value, connection, context):
+        if ((not value) and isinstance(value, six.string_types) and
+            connection.features.interprets_empty_strings_as_nulls):
+                return None
+        return value
+
+    def get_col(self, alias, source=None):
+        if source is None:
```

Comment:
Can `source` be falsey and not `None`? If not, I think it's better to use `source or self.related_field` and make the method a one-liner.